File: sdks/ruby/lib/hooksniff/inbound.rb
# frozen_string_literal: true

module HookSniff
  class Inbound
    def initialize(client)
      @client = client
    end

    def list_configs
      @client.request(:get, "/api/v1/inbound/configs")
    end

    def create_config(body)
      @client.request(:post, "/api/v1/inbound/configs", body: body)
    end

    def update_config(id, body)
      @client.request(:put, "/api/v1/inbound/configs/#{id}", body: body)
    end

    def delete_config(id)
      @client.request(:delete, "/api/v1/inbound/configs/#{id}")
    end
  end
end

File: sdks/csharp/HookSniff/Inbound.cs
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;

namespace HookSniff
{
    public class Inbound
    {
        private readonly HookSniffHttpClient _client;

        public Inbound(HookSniffHttpClient client)
        {
            _client = client;
        }

        public async Task<List<InboundConfigOut>> ListConfigsAsync(CancellationToken cancellationToken = default)
        {
            return await _client.GetAsync<List<InboundConfigOut>>("/api/v1/inbound/configs", cancellationToken);
        }

        public async Task<InboundConfigOut> CreateConfigAsync(InboundConfigIn body, CancellationToken cancellationToken = default)
        {
            return await _client.PostAsync<InboundConfigOut>("/api/v1/inbound/configs", body, cancellationToken);
        }

        public async Task<InboundConfigOut> UpdateConfigAsync(string id, InboundConfigIn body, CancellationToken cancellationToken = default)
        {
            return await _client.PutAsync<InboundConfigOut>($"/api/v1/inbound/configs/{id}", body, cancellationToken);
        }

        public async Task DeleteConfigAsync(string id, CancellationToken cancellationToken = default)
        {
            await _client.DeleteAsync($"/api/v1/inbound/configs/{id}", cancellationToken);
        }
    }

    public class InboundConfigOut
    {
        public string Id { get; set; }
        public string CustomerId { get; set; }
        public string Provider { get; set; }
        public string Secret { get; set; }
        public string EndpointId { get; set; }
        public bool Enabled { get; set; }
        public string CreatedAt { get; set; }
    }

    public class InboundConfigIn
    {
        public string Provider { get; set; }
        public string Secret { get; set; }
        public string EndpointId { get; set; }
        public bool? Enabled { get; set; }
    }
}
